packages/eslint-plugin/tests/lib/rules/boolean-literal-compare.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * @fileoverview Flags unnecessary equality comparisons against booleans
+ * @author Josh Goldberg <https://github.com/joshuakgoldberg>
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/boolean-literal-compare'),
+  RuleTester = require('eslint').RuleTester,
+  path = require('path');
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const rootDir = path.join(process.cwd(), 'tests/fixtures');
+const parserOptions = {
+  tsconfigRootDir: rootDir,
+  project: './tsconfig.json'
+};
+const ruleTester = new RuleTester({
+  parserOptions,
+  parser: '@typescript-eslint/parser'
+});
+
+ruleTester.run('boolean-literal-compare', rule, {
+  valid: [
+    `
+declare const varAny: any;
+declare const varString: string;
+declare const varObject: {};
+
+varAny === 'true';
+varAny === 'false';
+
+'false' === varString;
+'true' === varObject;
+
+varObject == varString;
+[] !== {};
+(() => "")() == (() => "");
```

Comment:
Please add a valid test case to confirm that this rule doesn't change the expression result. For example:

```ts
function f(a?: boolean): void {
    if (a === false) { // `a === false` and `!a` are different.
    }
}
```

Reply:
Ha, looks like that's a bug in the implementation of the rule. Will fix.